### DISCUSSION TITLE
400 Client Error: Bad Request for url

### DISCUSSION THREAD

**Discussion by starryloki** [open-webui/open-webui#1371] [General]
# Bug Report

## Description

**Bug Summary:**
I use one-api as a proxy aggregator for upstream APIs, which proxies OpenAI and Claude, and the downstream format of one-api is in OpenAI format. After accessing one-api through open-webui, I can use OpenAI normally but cannot use Claude. Although an error is prompted on open-webui, one-api has successfully requested Claude.
I once tried to use other tools to call Claude from one-api, and it was successful.

**Actual Behavior:**
External: {'message': 'messages.1.content.0.text.text: Field required (request id: 2024033123382364419163873473339)', 'type': 'invalid_request_error', 'param': '', 'code': None}

## Environment

- Debian 12 x86-64 6.1.0-18-amd64
- Safari lastest
- Open-WebUI docker lastest

## Reproduction Details

**Confirmation:**

- [Y] I have read and followed all the instructions provided in the README.md.
- [Y] I am on the latest version of both Open WebUI and Ollama.
- [Y] I have included the browser console logs.
- [Y] I have included the Docker container logs.

## Logs and Screenshots

**Docker Container Logs:**
```
INFO:     10.12.1.5:60299 - "POST /openai/api/chat/completions HTTP/1.1" 200 OK
INFO:     10.12.1.5:60299 - "POST /api/v1/chats/0dc7fe26-517e-4aa0-a775-f9aff7693411 HTTP/1.1" 200 OK
INFO:     10.12.1.5:60299 - "GET /api/v1/chats/ HTTP/1.1" 200 OK
INFO:     10.12.1.5:60299 - "POST /openai/api/chat/completions HTTP/1.1" 200 OK
INFO:     10.12.1.5:60299 - "POST /api/v1/chats/0dc7fe26-517e-4aa0-a775-f9aff7693411 HTTP/1.1" 200 OK
INFO:     10.12.1.5:60299 - "GET /api/v1/chats/ HTTP/1.1" 200 OK
INFO:     10.12.1.5:60299 - "GET /api/v1/chats/ HTTP/1.1" 200 OK
ERROR:apps.openai.main:400 Client Error: Bad Request for url: https://{API_ENDPOINT}/v1/chat/completions
Traceback (most recent call last):
  File "/app/backend/apps/openai/main.py", line 325, in proxy
    r.raise_for_status()
  File "/usr/local/lib/python3.11/site-packages/requests/models.py", line 1021, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 400 Client Error: Bad Request for url: https://{API_ENDPOINT}/v1/chat/completions
INFO:     10.12.1.5:60300 - "POST /openai/api/chat/completions HTTP/1.1" 500 Internal Server Error
ERROR:apps.openai.main:529 Server Error: status code 529 for url: https://{API_ENDPOINT}/v1/chat/completions
Traceback (most recent call last):
  File "/app/backend/apps/openai/main.py", line 325, in proxy
    r.raise_for_status()
  File "/usr/local/lib/python3.11/site-packages/requests/models.py", line 1021, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 529 Server Error: status code 529 for url: https://{API_ENDPOINT}/v1/chat/completions
INFO:     10.12.1.5:60300 - "POST /openai/api/chat/completions HTTP/1.1" 500 Internal Server Error
```
The request with port number 60299 is a request for gpt3.5, the request was successful, and all subsequent ones are requests for claude.
## Installation Method

docker

**Comment by tjbck** [comment]
I believe Claude API spec isn't compatible with OpenAI API spec. Try using LiteLLM for this use case: https://github.com/open-webui/open-webui/discussions/1038

**Reply by starryloki** [reply]
This can work successfully, but when LiteLLM is not configured, is it possible to request the Claude API in the form of an OpenAI API? one-api can gather multiple upstreams and issue them in the form of OpenAI.

**Reply by cheahjs** [reply]
`messages.1.content.0.text.text: Field required` sounds like one-api isn't translating the OpenAI API requests from Open WebUI correctly, and having a non-zero index on messages sounds like it's failing on a 2nd message (normally the assistant response).

If you can provide the payload of the call to `/openai/api/chat/completions`, I might be able to look into the issue for you, eg
```json
{"model":"Google: Gemma 7B","stream":true,"messages":[{"role":"user","content":"Tell me a random fun fact about the Roman Empire"}],"max_tokens":16000}
```

**Reply by starryloki** [reply]
Hello, @cheahjs , thank you for noticing this issue. I have captured the relevant payload:
```JSON
{"model": "claude-3-sonnet-20240229", "stream": true, "messages": [{"role": "user", "content": "test"}, {"role": "assistant", "content": ""}]}
```
I only noticed requests for /v1/chat/completions and no requests for /openai/api/chat/completions.

**Reply by cheahjs** [reply]
The empty assistant message is the problem, does this happen on a new chat?

**Reply by starryloki** [reply]
I started a new session, the payload and error are still consistent.I noticed that open-web will help me improve the title through upstream, and I found that title improvement can work normally.
Payload of Chat:
```JSON
{"model": "claude-3-sonnet-20240229", "stream": true, "messages": [{"role": "user", "content": "hi"}, {"role": "assistant", "content": ""}]}
```
Payload of title improvement:
```JSON
{"model": "claude-3-sonnet-20240229", "messages": [{"role": "user", "content": "Create a concise, 3-5 word phrase as a header for the following query, strictly adhering to the 3-5 word limit and avoiding the use of the word 'title': hi"}], "stream": false}
```

**Reply by justinh-rahb** [reply]
> This can work successfully, but when LiteLLM is not configured, is it possible to request the Claude API in the form of an OpenAI API? one-api can gather multiple upstreams and issue them in the form of OpenAI.

@starryloki so can LiteLLM, that's the entire reason we use it.

**Reply by starryloki** [reply]
Not a bad way :)
